calm_adapter/calm_api_client/src/test/scala/uk/ac/wellcome/platform/calm_api_client/CalmApiClientTest.scala

```diff
@@ -0,0 +1,155 @@
+package uk.ac.wellcome.platform.calm_api_client
+
+import java.time.LocalDate
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.{
+  Authorization,
+  BasicHttpCredentials,
+  Cookie,
+  RawHeader
+}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import uk.ac.wellcome.akka.fixtures.Akka
+import uk.ac.wellcome.fixtures.TestWith
+import uk.ac.wellcome.platform.calm_api_client.fixtures.{
+  CalmHttpTestClient,
+  CalmResponseGenerators
+}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CalmApiClientTest
+    extends AnyFunSpec
+    with Matchers
+    with ScalaFutures
+    with CalmResponseGenerators
+    with Akka {
+
+  val url = "calm.api"
+  val username = "calm-user"
+  val password = "calm-password"
+  val query = CalmQuery.ModifiedDate(LocalDate.of(2000, 1, 1))
+  val suppressedField = "Secret" -> "Shhhh"
+
+  it("performs search requests") {
+    val nResults = 10
+    val responses = List(searchResponse(nResults))
+    withApiClient(responses) {
+      case (apiClient, _) =>
+        whenReady(apiClient.search(query)) { response =>
+          response shouldBe CalmSession(nResults, Cookie(cookie))
+        }
+    }
+  }
+
+  it("performs summary requests") {
+    val responses = List(summaryResponse(List("RecordID" -> "1")))
+    withApiClient(responses) {
+      case (apiClient, _) =>
+        whenReady(apiClient.summary(1)) { response =>
+          response shouldBe CalmRecord(
+            "1",
+            Map("RecordID" -> List("1")),
+            retrievedAt)
+        }
+    }
+  }
+
+  it("uses basic auth credentials for requests") {
+    val responses =
+      List(searchResponse(1), summaryResponse(List("RecordID" -> "1")))
+    withApiClient(responses) {
+      case (apiClient, testHttpClient) =>
+        whenReady(for {
+          _ <- apiClient.search(query)
+          _ <- apiClient.summary(1)
+        } yield ()) { _ =>
+          testHttpClient.requests.map(_.headers.collect {
+            case auth: Authorization => auth
+          }) should contain only List(
+            Authorization(BasicHttpCredentials(username, password))
+          )
+        }
```

Comment:
This is all a bit smushed together; a couple of intermediate variables would help readability.

```suggestion
        val future = for {
          _ <- apiClient.search(query)
          _ <- apiClient.summary(1)
        } yield ()
        
        whenReady(future) { _ =>
          val authHeaders =
            testHttpClient.requests
              .flatMap { _.headers }
              .collect { case auth: Authorization => auth }

          authHeaders.distinct shouldBe List(
            Authorization(BasicHttpCredentials(username, password))
          )
        }
```